generators/client/templates/angular/src/main/webapp/app/shared/user/_user.service.ts

```diff
@@ -17,61 +17,48 @@
  limitations under the License.
 -%>
 import { Injectable } from '@angular/core';
-import { Http, Response } from '@angular/http';
+import { HttpClient, HttpResponse } from '@angular/common/http';
 import { Observable } from 'rxjs/Observable';
 
 import { SERVER_API_URL } from '../../app.constants';
-<%_ if (authenticationType !== 'oauth2') { _%>
 import { User } from './user.model';
-<%_ } _%>
-import { ResponseWrapper } from '../model/response-wrapper.model';
 import { createRequestOption } from '../model/request-util';
 
 @Injectable()
 export class UserService {
     private resourceUrl = SERVER_API_URL + '<%- apiUaaPath %>api/users';
 
-    constructor(private http: Http) { }
+    constructor(private http: HttpClient) { }
 <%_ if (authenticationType !== 'oauth2') { _%>
 
-    create(user: User): Observable<ResponseWrapper> {
-        return this.http.post(this.resourceUrl, user)
-            .map((res: Response) => this.convertResponse(res));
+    create(user: User): Observable<HttpResponse<User>> {
+        return this.http.post<User>(this.resourceUrl, user, { observe: 'response' });
     }
 
-    update(user: User): Observable<ResponseWrapper> {
-        return this.http.put(this.resourceUrl, user)
-            .map((res: Response) => this.convertResponse(res));
+    update(user: User): Observable<HttpResponse<User>> {
+        return this.http.put<User>(this.resourceUrl, user, { observe: 'response' });
     }
 
-    find(login: string): Observable<User> {
-        return this.http.get(`${this.resourceUrl}/${login}`).map((res: Response) => res.json());
+    find(login: string): Observable<HttpResponse<User>> {
+        return this.http.get<User>(`${this.resourceUrl}/${login}`, { observe: 'response' });
```

Comment:
@NilsWild after looking at the failing tests I'm wondering if we should return `HttpResponse` here, I kind of like the old code where we return the entity instead of the response as the components then doesn't have to worry about changes in service and can always expect to work with entities. Maybe we should keep the map method and do response.body in the services everywhere as it was before. What was the reason to change that behavior? I see that you removed almost all the map methods in services and did response.body in the components

Reply:
in some cases you referenced to status taht is not part of body but was part of reposnewrapper for consistency i wanted to retun httpresponse everywhere

Reply:
ah i see you returned responsewrapper in those cases like update but the entity directly in find

can change that back

Reply:
i'm also not sure if we should return response or just entity, I personally feel returning entity to be cleaner as we wont need headers or status in all cases, But i''ll let you choose as i'm not sure either

Reply:
on the one hand from an api point of view i feel like its inconsistent and confusing that the service does return different objects for the same general purpose (http communication) on the other hand returning just the api is also clean because the response is never used...

anyway i dont have time before tomorrow afternoon maybe there's a third opinion til then